fairscale/utils/reduce_scatter_bucketer.py

```diff
@@ -131,6 +150,7 @@ def flush(self) -> None:
         """Reduce-scatter any partial buckets."""
         for bucket in self.buckets.values():
             bucket.flush()
+            bucket.teardown()
```

Comment:
also, maybe we don't want to teardown on every flush? We might have lots of intermediate flush's during the backward pass, but we probably only want to teardown at the end. Perhaps expose teardown as a separate API and call it in the final backward hook in FSDP?